### PR TITLE
Custom separator edit route

### PR DESCRIPTION
**Pull request by RomanZavodskikh**
Change -edit-route and -clone-route to support arbitrary pattern separator

### CODE REVIEW

File: eskip/eskip.go [hunk 1]
```diff
@@ -32,16 +32,18 @@ var (
 //        r0: Source("127.0.0.1/8", "10.0.0.0/8") -> inlineContent("OK") -> <shunt>
 //        # actual route
 //        edit_r0: ClientIP("127.0.0.1/8", "10.0.0.0/8") -> inlineContent("OK") -> <shunt>
-func NewEditor(reg *regexp.Regexp, repl string) *Editor {
+func NewEditor(reg *regexp.Regexp, repl string, sep string) *Editor {
 	return &Editor{
 		reg:  reg,
 		repl: repl,
+		sep:  sep,
```
Comment (AlexanderYastrebov) [inline eskip/eskip.go:39]: Why do we need to know separator here?

File: config/routechangerconfig.go [hunk 2]
```diff
@@ -9,24 +9,27 @@ import (
...
 	if rcc.Reg == nil {
 		return ""
 	}
-	return fmt.Sprintf("/%s/%s/", rcc.Reg, rcc.Repl)
+	return fmt.Sprintf("%s%s%s%s%s", rcc.Sep, rcc.Reg, rcc.Sep, rcc.Repl, rcc.Sep)
 }
 
 func (rcc *routeChangerConfig) Set(value string) error {
-	a := strings.Split(value, "/")
+	firstSym := value[:1]
```
Comment (AlexanderYastrebov) [inline config/routechangerconfig.go:23]: We should check for empty `value` before, otherwise this would cause panic

File: eskip/eskip_test.go [hunk 3]
```diff
@@ -620,6 +628,15 @@ func TestEditorPreProcessor(t *testing.T) {
...
 			want:   r1FilterChanged,
+		},
+		{
+			name: "test that regex with / works properly",
+			rep: &Editor{
+				reg:  regexp.MustCompile("/26"),
+				repl: "/24",
+			},
+			routes: r2,
+			want:   r2Changed,
```
Comment (AlexanderYastrebov) [inline eskip/eskip_test.go:639]: I am not sure we need this test - it tests that regexp replacement works but it does not test the change this PR introduces

**Comment by AlexanderYastrebov** [comment]
It should be mentioned somewhere in the docs that first symbol of the value is used as a separator

File: eskip/eskip.go (hunk 1 above)
Comment (RomanZavodskikh) [reply eskip/eskip.go:39]: I don't see any obvious reason for separator here too... However, won't we need it for some debug/logging/etc purposes?

Comment (AlexanderYastrebov) [reply eskip/eskip.go:39]: The Editor does not have to know that there is any separator involved at all - it should only know about match and replacement values

**Comment by AlexanderYastrebov** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Comment by szuecs** [comment]
Thanks! Great first small change :)

**Pull request merged**